### PR TITLE
[BTS-277] Input

### PR DESCRIPTION
**Pull request by ddsilva**

### CODE REVIEW

File: components/Input/Input.jsx [hunk 1]
```diff
@@ -64,37 +25,31 @@ const InputIcon = styled(Icon)`
   bottom: 10px;
 `;
 
-const InputSearchIcon = styled(InputIcon)`
+const InputSearchIcon = styled(InputIcon).attrs({ name: 'search' })`
```
Comment (allyssonsantos) [inline components/Input/Input.jsx]: Why did you need this `name: 'search'`?

File: components/Input/Input.jsx [hunk 2]
```diff
@@ -64,37 +25,31 @@ const InputIcon = styled(Icon)`
...
-`;
-
-const RequiredMark = styled.em`
-  ${REQUIRED_MARK_STYLE}
+  ${({
+    theme: {
+      spacing: { small },
+    },
+  }) => `
+    padding: 0px ${small}px;
```
Comment (ggdaltoso) [inline components/Input/Input.jsx]:
```suggestion
    padding: 0 ${small}px;
```

File: components/Input/Input.jsx [hunk 3]
```diff
@@ -64,37 +25,31 @@ const InputIcon = styled(Icon)`
...
+        error: { 500: error500 },
+      },
+    },
+  }) => `color: ${error500};`}
 `;
 
 const DescriptionLabel = styled.span`
   cursor: text;
   display: block;
   font-size: 14px;
```
Comment (ggdaltoso) [inline components/Input/Input.jsx]: Why not `baseFontSize`?

File: components/Input/Input.jsx [hunk 4]
```diff
@@ -64,37 +25,31 @@ const InputIcon = styled(Icon)`
   bottom: 10px;
 `;
 
-const InputSearchIcon = styled(InputIcon)`
+const InputSearchIcon = styled(InputIcon).attrs({ name: 'search' })`
   left: 12px;
```
Comment (ggdaltoso) [inline components/Input/Input.jsx]: Could you add the `spacing.small` here?

File: components/Input/Input.jsx [hunk 5]
```diff
@@ -64,37 +25,31 @@ const InputIcon = styled(Icon)`
   bottom: 10px;
```
Comment (ggdaltoso) [inline components/Input/Input.jsx]: Turn this into a themeable variant

File: components/Input/sub-components/TextInput.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      &[disabled] {
+        background-color: ${neutral300};
+        border-color: ${neutral500};
+        box-shadow: none;
+        color: ${neutral500};
+        cursor: not-allowed;
+      }
+
+      :-webkit-autofill {
+        box-shadow: 0 0 0px 1000px ${error100} inset;
```
Comment (ggdaltoso) [inline components/Input/sub-components/TextInput.jsx]:
```suggestion
        box-shadow: 0 0 0 1000px ${error100} inset;
```

File: components/shared/ErrorMessage.jsx [hunk 7]
```diff
@@ -2,7 +2,7 @@ import styled from 'styled-components';
 import PropTypes from 'prop-types';
 import { colors } from './theme';
 
-const ErrorMessage = styled.div`
+const ErrorMessage = styled.span`
   color: ${({
     theme: {
       colors: { error },
```
Comment (ggdaltoso) [inline components/shared/ErrorMessage.jsx]:
```suggestion
      colors: { error: { 900: error900 } },
```

File: components/Input/Input.jsx (hunk 1 above)
Comment (ddsilva) [reply components/Input/Input.jsx]: Because it's an Icon component, so it needs this prop to render a "search" one. :)

File: components/Input/Input.jsx (hunk 3 above)
Comment (ddsilva) [reply components/Input/Input.jsx]: :heavy_check_mark:

File: components/Input/Input.jsx (hunk 4 above)
Comment (ddsilva) [reply components/Input/Input.jsx]: :heavy_check_mark:

File: components/Input/Input.jsx (hunk 5 above)
Comment (ddsilva) [reply components/Input/Input.jsx]: :heavy_check_mark:

File: components/Input/Input.jsx (hunk 2 above)
Comment (ddsilva) [reply components/Input/Input.jsx]: :heavy_check_mark:

File: components/shared/ErrorMessage.jsx (hunk 7 above)
Comment (ddsilva) [reply components/shared/ErrorMessage.jsx]: :heavy_check_mark:

File: components/Input/sub-components/TextInput.jsx (hunk 6 above)
Comment (ddsilva) [reply components/Input/sub-components/TextInput.jsx]: :heavy_check_mark:
